sources/packages/backend/apps/api/src/route-controllers/student/student.external.controller.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  Body,
+  Controller,
+  HttpCode,
+  HttpStatus,
+  NotFoundException,
+  Post,
+} from "@nestjs/common";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  RequiresUserAccount,
+} from "../../auth/decorators";
+import BaseController from "../BaseController";
+import { StudentService } from "@sims/integrations/services";
+import {
+  ExternalSearchStudentAPIInDTO,
+  StudentDetailsAPIOutDTO,
+} from "./models/student.dto";
+
+/**
+ * Student controller for external client.
+ */
+@RequiresUserAccount(false)
+@AllowAuthorizedParty(AuthorizedParties.external)
```

Comment:
Adding the comment here to facilitate the discussion.
Can we please ensure that the token scopes that are not needed and also the non-needed mappers will be removed from the token?
The token should have only the necessary information.

For instance, can we remove the below.
![image](https://github.com/user-attachments/assets/38f16eb9-0acd-4e0b-b779-1f8695eb0f83)

Reply:
I removed all the other mappers that automatically come from the client and in `<client-id>-dedicated` there are some mappers (clientHost, clientAddress and clientId). I tried to remove them, if you refresh the page, they're still there. I think they are not optional.
![image](https://github.com/user-attachments/assets/9cd3f2f9-1e0c-4f5e-b412-acdaf28a76f0)

Reply:
Even trying to remove `<client-id>-dedicated` is not allowed:
![image](https://github.com/user-attachments/assets/7496f51d-5367-436b-a465-0a3a9a856336)